### PR TITLE
[SHELL32] Folder options: Enable visual styles for RunDLL

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Proposed changes
- Enable visual styles for RunDLL using ~ReactOS default DLL manifest~ manifest resource ID 123.
- Dialog: Use `MAKEINTRESOURCE` instead of `LoadString`

Before:
ReactOS_Before

After:
ReactOS_After

### CODE REVIEW

**Comment by HBelusca** [comment]
I wonder why the visual styles weren't enabled by default already. Is it because shell32 dll itself doesn't incorporate the manifest? Or because the control.exe app https://git.reactos.org/?p=reactos.git;a=tree;f=base/applications/control;hb=a0617472955d3ce012166f62fcee8a0f349f60d9  does not?

**Review by HBelusca — COMMENTED** [review]
It may be more interesting to do this sxs activation within the `ShowFolderOptionsDialogThreadProc()` worker function instead.

**Comment by TAN-Gaming** [comment]
> I wonder why the visual styles weren't enabled by default already. Is it because shell32 dll itself doesn't incorporate the manifest? Or because the control.exe app https://git.reactos.org/?p=reactos.git;a=tree;f=base/applications/control;hb=a0617472955d3ce012166f62fcee8a0f349f60d9 does not?

shell32.dll itself does have the manifest in resource, but it doesn't activated by the time DLL is loaded.
[https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/shell32.rc;h=7103f8005406cdd8b485e59d4ab6ec5a7f33d780;hb=a0617472955d3ce012166f62fcee8a0f349f60d9#l80](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/shell32.rc;h=7103f8005406cdd8b485e59d4ab6ec5a7f33d780;hb=a0617472955d3ce012166f62fcee8a0f349f60d9#l80)

Our control.exe calls explorer/rundll32 internally and doesn't host the DLL itself so it doesn't need a manifest.

**Comment by TAN-Gaming** [comment]
> It may be more interesting to do this sxs activation within the `ShowFolderOptionsDialogThreadProc()` worker function instead.

Explorer also uses this function to open the folder options dialog. In this case, sxs activation would be unnecessary.

**Comment by TAN-Gaming** [comment]
> It may be more interesting to do this sxs activation within the `ShowFolderOptionsDialogThreadProc()` worker function instead.

Explorer also use this function to open the folder options dialog. In this case, sxs activation would be unnecessary.

**Review by oleg-dubinskiy — APPROVED** [review]
Just for comparison, on XP SP3 at least, this dialog is also themed. So yes, this matches Windows behaviour. LGTM. :+1: 
folder-options-themed

**Review by cbialorucki — APPROVED** [review]

**Comment by binarymaster** [comment]
@TAN-Gaming I'd recommend associating the PR commit with this ticket on merge: [CORE-12285](https://jira.reactos.org/browse/CORE-12285)

**Comment by TAN-Gaming** [comment]
> @TAN-Gaming I'd recommend associating the PR commit with this ticket on merge: [CORE-12285](https://jira.reactos.org/browse/CORE-12285)

OK.

File: dll/win32/shell32/dialogs/folder_options.cpp [hunk 1]
```diff
@@ -308,6 +304,20 @@ ShowFolderOptionsDialog(UINT Page, BOOL Async = FALSE)
...
+    ULONG_PTR Cookie;
+    BOOL bActivated;
+
+    // Enable visual styles
+    ZeroMemory(&ActCtx, sizeof(ActCtx));
+    ActCtx.cbSize = sizeof(ActCtx);
+    ActCtx.dwFlags = ACTCTX_FLAG_HMODULE_VALID | ACTCTX_FLAG_RESOURCE_NAME_VALID;
+    ActCtx.lpResourceName = MAKEINTRESOURCEW(ISOLATIONAWARE_MANIFEST_RESOURCE_ID); // Default DLL manifest
+    ActCtx.hModule = shell32_hInstance;
+    hActCtx = CreateActCtxW(&ActCtx);
```
Comment (learn-more) [inline dll/win32/shell32/dialogs/folder_options.cpp:318]: Is this manual manifest activation also needed on Windows?

Comment (TAN-Gaming) [reply dll/win32/shell32/dialogs/folder_options.cpp:318]: Windows (XPSP3) calls explorer.exe to open the dialog and doesn't use rundll32.exe to host the dll.

**Comment by TAN-Gaming** [comment]
Needs some tests on Windows.

File: dll/win32/shell32/dialogs/folder_options.cpp (hunk 1 above)
Comment (TAN-Gaming) [reply dll/win32/shell32/dialogs/folder_options.cpp]: Tested on Win2k3. This is not a bug in our loader. Calling the folder options dialog from ReactOS shell32.dll on Win2k3 has the visual style issue as well.

Command: `rundll32.exe <path_to_reactos_shell32_dll>,Options_RunDLL 0`

**Comment by TAN-Gaming** [comment]
RunDLL command: `rundll32.exe <path_to_reactos_shell32_dll>,Options_RunDLL 0`

ReactOS shell32.dll on Win2k3 (before):
ReactOS_Shell32_Win2k3_Before

ReactOS shell32.dll on Win2k3 (after):
ReactOS_Shell32_Win2k3_After

ReactOS (after):
VirtualBox_ReactOS_17_06_2024_21_10_04

Resource ID 123 is documented here:
[https://learn.microsoft.com/en-us/windows/win32/controls/cookbook-overview?redirectedfrom=MSDN#using-comctl32-version-6-in-control-panel-or-a-dll-that-is-run-by-rundll32exe](https://learn.microsoft.com/en-us/windows/win32/controls/cookbook-overview?redirectedfrom=MSDN#using-comctl32-version-6-in-control-panel-or-a-dll-that-is-run-by-rundll32exe)

**Review by JoachimHenze — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
You can add the referenced link (re. Resource ID 123) to the second commit message.

**Review by katahiromz — APPROVED** [review]

**Comment by TAN-Gaming** [comment]
Let's go.

**Pull request merged**